app/src/main/java/de/westnordost/streetcomplete/quests/note_discussion/NoteDiscussionForm.java

```diff
@@ -162,11 +229,61 @@ private void onClickOk()
 
 		Bundle answer = new Bundle();
 		answer.putString(TEXT, noteText);
+		answer.putStringArrayList(IMAGE_PATHS, imagePaths);
 		applyImmediateAnswer(answer);
 	}
 
+	private void takePhoto()
+	{
+		Intent takePhotoIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+		if (takePhotoIntent.resolveActivity(getActivity().getPackageManager()) != null) {
+			try {
+				photoFile = createImageFile();
+			} catch (IOException e) {
+			}
+			if (photoFile != null) {
+				if (Build.VERSION.SDK_INT > 21) {
+					takePhotoIntent.putExtra(MediaStore.EXTRA_OUTPUT, getUriForFile(getActivity(), "de.westnordost.streetcomplete.fileprovider", photoFile));
+				} else {
+					takePhotoIntent.putExtra(MediaStore.EXTRA_OUTPUT, Uri.fromFile(photoFile));
+				}
+				startActivityForResult(takePhotoIntent, REQUEST_TAKE_PHOTO);
+			}
+		}
+	}
+
+	@Override
+	public void onActivityResult(int requestCode, int resultCode, Intent data) {
+		if (requestCode == REQUEST_TAKE_PHOTO) {
+			if (resultCode == RESULT_OK)
+			{
+				imageBitmaps.add(BitmapFactory.decodeFile(photoFile.getAbsolutePath()));
+				imagePaths.add(photoFile.toString());
+			} else
+			{
+				if (photoFile.exists())
+				{
+					photoFile.delete();
+				}
+			}
+		}
+	}
+
+	private File createImageFile() throws IOException
+	{
+		String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
+		String imageFileName = "JPEG_" + timeStamp + "_";
+		File directory = getActivity().getExternalFilesDir(Environment.DIRECTORY_PICTURES);
```

Comment:
Doesn't external files dir always require the read device storage permission? The photo should be saved into the app's private storage. (As far as I remember: `getFilesDir`)

Reply:
The documentation says:
> However, if you'd like the photos to remain private to your app only, you can instead use the directory provided by ```getExternalFilesDir()```. On Android 4.3 and lower, writing to this directory also requires the ```WRITE_EXTERNAL_STORAGE``` permission. Beginning with Android 4.4, the permission is no longer required because the directory is not accessible by other apps, so you can declare the permission should be requested only on the lower versions of Android by adding the ```maxSdkVersion``` attribute

Reply:
Ok